fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/ApiRequestImpl.java

```diff
@@ -385,6 +385,27 @@ protected void validateMetrics(Set<LogicalMetric> logicalMetrics, LogicalTable t
             String apiIntervalQuery,
             Granularity granularity,
             DateTimeFormatter dateTimeFormatter
+    ) throws BadApiRequestException {
+        return generateIntervals(new DateTime(), apiIntervalQuery, granularity, dateTimeFormatter);
+    }
+
+
+    /**
+     * Extracts the set of intervals from the api request.
```

Comment:
How is this version of the method different from the one that doesn't take "now"? Why would the caller choose to use one over the other?